provisioner/clusterpy.go

```diff
@@ -106,7 +106,7 @@ func (p *clusterpyProvisioner) Supports(cluster *api.Cluster) bool {
 	return cluster.Provider == providerID
 }
 
-func (p *clusterpyProvisioner) updateDefaults(cluster *api.Cluster, channelConfig channel.Config, adapter *awsAdapter) error {
+func (p *clusterpyProvisioner) updateDefaults(cluster *api.Cluster, channelConfig channel.Config, adapter *awsAdapter, instanceTypes *awsUtils.InstanceTypes) error {
```

Comment:
I think `InstanceInfo` could be a method of awsAdapter (it can cache result of API call if needed).

Also I now see there is a whole `type Instance` that provides access to instance properties with similar accessors like `MemoryFraction`, see https://github.com/zalando-incubator/kubernetes-on-aws/blob/6f3c355069f1aebda0f77de6d831612f7d33c001/cluster/node-pools/master-default/userdata.yaml#L206

I think we should unify it, maybe by providing all instance types in the values and re-using existing accessors (add another for CPU):
```yaml
memory: {{ .Values.InstanceInfos["c6g-xlarge"].MemoryFraction 70 }}
```
or migrate `.Values.InstanceInfo.MemoryFraction` to proposed new template functions.

Reply:
I don't have a strong opinion on this. In the past we just needed to access the current instance in the context of the node pool, so no need to make it more elaborate. Now that we have more use cases I think it's fine to change it. Just keep in mind the migration path, we need to support both until the equivalent kubernetes-on-aws changes would be merged to all channels.

Reply:
Yes, I had something similar in mind. We can add the new functions first, roll them out and use them in one VPA to test. Once it runs fine through all channels I can create a follow-up ticket to migrate to the new functions and once the migration finishes, we can remove the old ones.

Reply:
I created a new issue for this specific problem. The change in this PR is already quite sizeable and I suggest we move on with this change, propagate it through the channels and then once it's finished, we can move towards the refactoring part.

I'll resolve this conversation now, we can carry on in the new issue, on this topic: https://github.bus.zalan.do/teapot/issues/issues/3566

I'll just proceed to test the feature-complete change in the current state so we can move forward.